### PR TITLE
Removed support for type string from `is`

### LINKED ISSUE #22: Remove support for type string from `is`
In the test, there are rules, like

```js
expect(is(function(){}, 'factory')).to.eq(false)
expect(is(function(x){}, 'factory')).to.eq(true)
expect(is(function(x,y){}, 'factory')).to.eq(false)
```

1. The number of parameters on a function should never give any indications about the intention of a function. The intent of a function is implementation-specific, not declaration-specific.
2. There can be factories with multiple or no arguments at all!

### PR DESCRIPTION
**Pull request by Download**
Fixes #22 

@minecrawler  Can you let me know what you think?

(I changed one ESLint setting... Just got rid of some blank lines.. Hope you can live with it?)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -148,14 +125,26 @@ function getInterface(proto) {
...
     return results
 }
 
+function isMixin(x) {
+    return (typeof x == 'function') && !!x.mixin
+}
+
+function isClass(x) {
+    return (typeof x == 'function') && (s => /^class\s/.test(s)
+        || /^.*classCallCheck\(/.test(s.replace(/^[^{]*{\s*/,'').replace(/\s*}[^}]*$/,'')))(x.toString())
```
Comment (minecrawler) [inline src/index.js]: What purpose does that arrow function serve? Why not assign `const s = x.toString()` and execute that stuff directly in the scope of `isClass()`?

Also, imho, creating such a long expression is hard to read. If this is about saving bytes, then leave it to Babel :) Keep the source clean and easy to read and let the transpiler do the dirty work!

**Comment by minecrawler** [comment]
As for the linting: If that's what you want, I can't stop you. I am only concerned about the readability for other developers.

Other than that: Could you try to separate changes you do into several commit in the future? It keeps the history expressive and documentary.

**Review by minecrawler — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (Download) [reply src/index.js]: Ok but that is a different issue. This was about removing the type strings (`'class'`, `'factory'` etc) support from `is()` and `like()`.

Comment (Download) [reply src/index.js]: Do you want me to change this for this PR?

**Comment by Download** [comment]
> As for the linting: If that's what you want, I can't stop you.

Mmm... That's not good.

I was hoping you got the feeling that, yes, you can stop me.
We are in this together.

I sincerely feel the extra whitespace reduces readability... Maybe it's because I have a widescreen monitor that is not FullHD... so I don't have so much vertical space... too much scrolling.

But my monitor should not dictate style guides. So please let me know and I can change it back if it bothers you.

File: src/index.js (hunk 1 above)
Comment (minecrawler) [reply src/index.js]: > Ok but that is a different issue.

it's new code in this PR, so it is directly related ;)

> Do you want me to change this

yeah. If you cannot give a decent explanation, just [KISS](https://en.wikipedia.org/wiki/KISS_principle).

**Comment by minecrawler** [comment]
> widescreen monitor that is not FullHD

oh, I see how your style developed (it's probably comparable to my laptop screen, which I cannot work with)! Sitting here with my two FullHD 16:9 monitors, of which one is rotated by 90° in order to see more code. For me ([and many other devs](https://hashnode.com/post/how-many-monitors-do-you-guys-use-to-codedesignwork-ciwvsi52q00u08k53bcujxx6x)), having many big monitors is quite natural. Your and my world are hard to compare, I guess.

Though because of above reasons, usually "clean code" is code which unfolds vertically, adding logical blocks (like grouping variable declarations) and full code constructs (like `if() {}` over at least three lines) instead of shortcuts (like `c = a && b`). Writing very stable, distinct code is what should be the driver (hence the preference for `;`). To sum it up, code which would force you (personally) to type and scroll a lot.

Imho, while general coding standards are a good thing, I can't have you as the project owner and main developer get unmotivated, because the project becomes tedious for you. So, while I still think that some changes to a few of the overly-long statements should be considered, let's try to keep vertical white spaces 
 and vertically unfolded expressions to a minimum for now.

On that note, maybe I can also introduce you to [Hashnode](https://hashnode.com/), which is a very nice and welcoming developer community to discuss tech and techniques, guidelines and practices. People also share many useful links about the latest developments, so you stay up2date as a plus. If you are interested, I would be delighted to see you around :)

File: src/index.js (hunk 1 above)
Comment (Download) [reply src/index.js]: I will change it.  

The explanation is that it's a trick to avoid having to declare a new variable. It just saves some bytes.

Comment (minecrawler) [reply src/index.js]: > It just saves some bytes.

KISS, make sure it is readable. The rest should be handled by the transpiler. Also `const s =` vs `=>()` is saving two bytes at the expense of performance (the interpreter has to allocate a new closure) and readability. I feel that our internet today is fast enough to handle those two bytes, so ultimately you save transfer speed, but pay double during execution... every time that line is executed. It's hard to balance those things, but I think you should favor runtime performance in those cases

**Comment by Download** [comment]
Hashnode looks interesting! Thanks for the tip.

**Review by minecrawler — APPROVED** [review]
LGTM

**Pull request merged**